.github/workflows/deploy.yml

```diff
@@ -408,19 +409,56 @@ jobs:
       - name: Install Kurtosis CDK tools
         uses: ./.github/actions/setup-kurtosis-cdk
 
+      # Step to handle when the SP1_PRIVATE_KEY secret is available.
+      - name: Replace Agglayer SP1 Prover Key with Github Secrets
+        run: |
+          sed -i "s/agglayer_prover_sp1_key: \"\"/agglayer_prover_sp1_key: \"${{ secrets.SP1_PRIVATE_KEY }}\"/" ./.github/tests/agglayer-sp1-key.yml
+        # This step will only execute if the necessary secrets are available, preventing failures
+        # on pull requests from forked repositories.
+        if: ${{ env.agglayer_prover_sp1_key && env.agglayer_prover_sp1_key != '' }}
+        env :
+          agglayer_prover_sp1_key: ${{ secrets.SP1_PRIVATE_KEY }}
+      
+      # Step to handle when the SP1_PRIVATE_KEY secret is not available.
+      - name: Handle missing SP1_PRIVATE_KEY secret
+        if: ${{ !env.agglayer_prover_sp1_key || env.agglayer_prover_sp1_key == '' }}
+        run: |
+          sed -i '/agglayer_prover_sp1_key:/d' ./.github/tests/agglayer-sp1-key.yml
+        env:
+          agglayer_prover_sp1_key: ${{ secrets.SP1_PRIVATE_KEY }}
```

Comment:
Also curious why did you need this step? The secret `SP1_PRIVATE_KEY` is available in this repository but you may not have access when running the PR from your fork. A work-around could be to create a PR in your own fork repository, add the secret and check that the CI passes.